compiler/plugins/input/Torch/PluginRegistration.cpp

```diff
@@ -57,8 +57,12 @@ struct TorchSession
       OpPassManager &passManager, std::string_view typeMnemonic) override {
     if (typeMnemonic == "onnx") {
       // ONNX input is a pre-processing step to torch.
-      passManager.addNestedPass<func::FuncOp>(
-          mlir::torch::onnx_c::createTorchOnnxToTorchPass());
+      mlir::torch::Torch::TorchLoweringPipelineOptions torchOnnxPipelineOptions;
+      torchOnnxPipelineOptions.backendLegalOps = {"aten.flatten.using_ints",
+                                                  "aten.unflatten.int"};
+      torchOnnxPipelineOptions.decompose = true;
```

Comment:
This could also use a comment.

For my own understanding, can you give an example of what this decompose option will do? I see that it adds `createDecomposeComplexOpsPass`. Maybe give an example of such an op and explain why IREE doesn't have its own lowering path for that op?

In other pipelines we sometimes leave the decision to decompose or not to each compilation target (e.g. llvm-cpu or vulkan-spirv), since certain ops may be implemented more efficiently/directly using those targets, without needing to decompose into a slower sequence of ops.

Reply:
The decomposition is mostly conditional on what mid-level IR torch-mlir is lowering to (StableHLO, linalg, or Tosa). Since we are always lowering to linalg in IREE, this should be standardized for the ONNX FE path.

Reply:
I don't think we need to initialize that op here, since it's default value is True, and that's the behavior we want.